ledger/sandbox-on-x/src/main/scala/com/daml/ledger/sandbox/bridge/validate/DeduplicationState.scala

```diff
@@ -22,25 +21,28 @@ case class DeduplicationState private (
       commandDeduplicationDuration: Duration,
       recordTime: Time.Timestamp,
   ): (DeduplicationState, Boolean) = {
+    assert(
```

Comment:
not this PR, performance nitpick: the undelying map here relies on efficiency of the ChangeId hash-code and equals, which is using a homegrown hashing tools to generate lazy hash codes. The homegrown might be not that efficient, and the lazy access is slower (since you are building a map from it, you will need this anyway)

Reply:
You are right that we have the lazy there. I've used that definition since it is the reference implementation for a `ChangeId`. I could use my own but with a cost: to be decoupled from what other parts of the code are using. What happens if the the formal `ChangeId` changes and welcomes one more field? That would leave this part of the code possible stale. I am curious what is the performance impact here, I'll try to benchmark it.